src/main/java/org/kuali/kra/award/home/Award.java

```diff
@@ -2885,20 +2885,52 @@ public void setActivityType(ActivityType activityType) {
 
     /**
      * This method removes Funding Proposal for specified index from list
-     * 
+     *
      * It also removes the AwardFundingProposal from the InstitutionalProposal
-     * 
+     *
      * @param index
      */
     public AwardFundingProposal removeFundingProposal(int index) {
-        AwardFundingProposal afp = (index >= 0) ? fundingProposals.remove(index) : null;
-        afp.getProposalId();
-        InstitutionalProposal proposal = getInstitutionalProposalService().getInstitutionalProposal(afp.getProposalId().toString());
+        AwardFundingProposal afp = (index >= 0) ? getFundingProposals().get(index) : null;
+
+        if (afp != null) {
+            removeFundingProposal(afp);
+        }
+
+        return afp;
+    }
+
+    /**
+     * Removes any funding proposals associated with the provided proposal number
+     *
+     * @param proposalNumber The proposal number of the IP version(s) to be removed
+     */
+    public List<AwardFundingProposal> removeFundingProposals(String proposalNumber) {
+        List<AwardFundingProposal> fundingProposals = getFundingProposals();
+        List<AwardFundingProposal> collectedFundingProposals = new ArrayList<AwardFundingProposal>();
+
+        for (AwardFundingProposal afp : fundingProposals) {
+            if (afp.getProposal() != null && StringUtils.equals(afp.getProposal().getProposalNumber(), proposalNumber)) {
+                collectedFundingProposals.add(afp);
+            }
+        }
+
+        for (AwardFundingProposal cfp : collectedFundingProposals) {
+            removeFundingProposal(cfp);
+        }
+
+        return fundingProposals;
+    }
+
+    public void removeFundingProposal(AwardFundingProposal afp) {
+        fundingProposals.remove(afp);
+        //allFundingProposals.remove(afp);

```

Comment:
What is the reason why we removed this commented line out of the foundation's fix?

Reply:
Never mind, found the comment in the Jira:
"Did not use the line "allFundingProposals.remove(afp);", this does not exist in our UAR version."
We'll give it a try without this...